src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsintegrationapi/services/GetPersonServiceTest.kt

```diff
@@ -25,59 +25,107 @@ internal class GetPersonServiceTest(
   @MockBean val probationOffenderSearchGateway: ProbationOffenderSearchGateway,
   private val getPersonService: GetPersonService
 ) : DescribeSpec({
-  val id = "abc123"
 
   beforeEach {
     Mockito.reset(nomisGateway)
     Mockito.reset(prisonerOffenderSearchGateway)
     Mockito.reset(probationOffenderSearchGateway)
   }
 
-  it("retrieves a person from NOMIS") {
-    getPersonService.execute(id)
+  describe("by search criteria") {
+    val firstName = "Bruce"
+    val lastName = "Wayne"
 
-    verify(nomisGateway, VerificationModeFactory.times(1)).getPerson(id)
-  }
+    it("returns person(s) from Prisoner Offender Search") {
+      getPersonService.execute(firstName, lastName)
 
-  it("retrieves a person from Prisoner Offender Search") {
-    getPersonService.execute(id)
+      verify(prisonerOffenderSearchGateway, VerificationModeFactory.times(1)).getPersons(firstName, lastName)
+    }
 
-    verify(prisonerOffenderSearchGateway, VerificationModeFactory.times(1)).getPerson(id)
-  }
+    it("returns person(s) from Probation Offender Search") {
+      getPersonService.execute(firstName, lastName)
 
-  it("retrieves a person from Probation Offender Search") {
-    getPersonService.execute(id)
+      verify(probationOffenderSearchGateway, VerificationModeFactory.times(1)).getPersons(firstName, lastName)
+    }
 
-    verify(probationOffenderSearchGateway, VerificationModeFactory.times(1)).getPerson(id)
-  }
+    it("returns person(s)") {
+      val personsFromPrisonerOffenderSearch = listOf(
+        Person("Bruce", "Wayne", middleName = "Gary")
+      )
+      val personsFromProbationOffenderSearch = listOf(
+        Person("Bruce", "Wayne", middleName = "John")
+      )
 
-  it("returns a person") {
-    val personFromNomis = Person("Billy", "Bob")
-    val personFromPrisonerOffenderSearch = Person("Sally", "Sob")
-    val personFromProbationOffenderSearch = Person("Molly", "Mob")
+      whenever(prisonerOffenderSearchGateway.getPersons(firstName, lastName)).thenReturn(personsFromPrisonerOffenderSearch)
+      whenever(probationOffenderSearchGateway.getPersons(firstName, lastName)).thenReturn(personsFromProbationOffenderSearch)
 
-    whenever(nomisGateway.getPerson(id)).thenReturn(personFromNomis)
-    whenever(prisonerOffenderSearchGateway.getPerson(id)).thenReturn(personFromPrisonerOffenderSearch)
-    whenever(probationOffenderSearchGateway.getPerson(id)).thenReturn(personFromProbationOffenderSearch)
+      val expectedResult = listOf(
+        Person(firstName, lastName, middleName = "Gary"),
+        Person(firstName, lastName, middleName = "John")
+      )
 
-    val result = getPersonService.execute(id)
+      val result = getPersonService.execute(firstName, lastName)
+      result.shouldBe(expectedResult)
```

Comment:
I think this can be simplified to:

```kotlin
it("returns person(s)") {
  val personsFromPrisonerOffenderSearch = listOf(Person(firstName, lastName, middleName = "Gary"))
  val personsFromProbationOffenderSearch = listOf(Person(firstName, lastName, middleName = "John"))

  whenever(prisonerOffenderSearchGateway.getPersons(firstName, lastName)).thenReturn(personsFromPrisonerOffenderSearch)
  whenever(probationOffenderSearchGateway.getPersons(firstName, lastName)).thenReturn(personsFromProbationOffenderSearch)

  val result = getPersonService.execute(firstName, lastName)

  result.shouldBe(listOf(personsFromPrisonerOffenderSearch, personsFromProbationOffenderSearch))
}
```